crates/revm/src/db/states/bundle_state.rs

```diff
@@ -130,9 +134,17 @@ impl BundleState {
             state,
             contracts: contracts.into_iter().collect(),
             reverts,
+            reverts_size,
         }
     }
 
+    /// Returns the approximate size of changes in the bundle state.
+    /// The estimation is not precise, because the information about the number of
+    /// destroyed entries that need to be removed is not accessible to the bundle state.
+    pub fn size_hint(&self) -> usize {
+        self.state.len() + self.contracts.len() + self.reverts_size
```

Comment:
state can have both the account and the storage. Maybe we should take that into consideration. But it gets harder to update

Reply:
Let us be okay with not ideal situation, as adding storage counting of state can get complex and can result in a fault code